File: src/main/scala/com/vc/easy/L101.scala
package com.vc.easy

class TreeNode(var _value: Int) {
  var value: Int = _value
  var left: TreeNode = null
  var right: TreeNode = null
}

object Solution {
  def isSymmetric(root: TreeNode): Boolean = {
    if(root == null) true
    else isSameTree(root.left, root.right)
  }

  def isSameTree(r1: TreeNode, r2: TreeNode): Boolean = {
    if(r1 != null && r2 != null) {
      r1.value == r2.value && isSameTree(r1.left, r2.right) && isSameTree(r1.right, r2.left)
    }
    else if(r1 == null && r2 == null) true
    else false
  }
}

File: src/main/java/com/vc/easy/L101.java
package com.vc.easy;

import com.vc.hard.TreeNode;

import java.util.*;

class L101 {
    public boolean isSymmetric(TreeNode root) {
        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);
        q.offer(root);
        while(!q.isEmpty()) {
            TreeNode t1 = q.poll();
            TreeNode t2 = q.poll();

            if(t1 == null ^ t2 == null) return false;
            if(t1 == null && t2 == null) continue;

            if(t1.val != t2.val) return false;

            q.offer(t1.left);
            q.offer(t2.right);
            q.offer(t1.right);
            q.offer(t2.left);
        }
        return true;
    }
}
